dev/clint/src/clint/linter.py

```diff
@@ -276,6 +276,42 @@ def _find_artifact_path_index(call_path: tuple[str, str]) -> int | None:
     return None
 
 
+class TypeAnnotationVisitor(ast.NodeVisitor):
```

Comment:
I think we should introduce a type checker like pyright instead but the codebase is not ready for that.